### PR TITLE
[netcpp] Update to 0.4.1

### PR DESCRIPTION
**Pull request by index1207**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by index1207** [comment]
@MonicaLiu0311  In my environment, I successfully installed this package on Linux with local vcpkg. Can you show me the error log?

**Comment by dg0yt** [comment]
Download CI log artifacts at
https://dev.azure.com/vcpkg/public/_build/results?buildId=106304&view=artifacts&pathAsName=false&type=publishedArtifacts

~~~
FAILED: src/CMakeFiles/netcpp.dir/native.cpp.o 
/usr/bin/c++ -DNETCPP_BUILD -D_LIBCPP_ENABLE_EXPERIMENTAL -I/mnt/vcpkg-ci/b/netcpp/src/v0.4.1-25bd05a929.clean/src/include -I/mnt/vcpkg-ci/b/netcpp/src/v0.4.1-25bd05a929.clean/include -fPIC -g -std=gnu++20 -MD -MT src/CMakeFiles/netcpp.dir/native.cpp.o -MF src/CMakeFiles/netcpp.dir/native.cpp.o.d -o src/CMakeFiles/netcpp.dir/native.cpp.o -c /mnt/vcpkg-ci/b/netcpp/src/v0.4.1-25bd05a929.clean/src/native.cpp
In file included from /mnt/vcpkg-ci/b/netcpp/src/v0.4.1-25bd05a929.clean/src/native.cpp:1:
/mnt/vcpkg-ci/b/netcpp/src/v0.4.1-25bd05a929.clean/include/net/native.hpp:30:10: fatal error: liburing.h: No such file or directory
   30 | #include <liburing.h>
      |          ^~~~~~~~~~~~
~~~

Look like it lacks the `-I...` for using vcpkg's liburing. 
And actually I don't see any relevant lookup in `CMakeLists.txt`. You will have to patch it. liburing provides pkg-config files.

**Comment by index1207** [comment]
The CI has passed. But it still couldn't find {TRIPLET}/lib/liburing/liburing.pc file. How do I modify it?
error log:
```
/usr/bin/ld: cannot find -luring: No such file or directory
```

**Comment by index1207** [comment]
I'm trying to create a new app with vcpkg manifest and link this package. and find liburing.pc and add cmake command `link_directories`. But still it makes `-luring` not found error.

**Comment by index1207** [comment]
> I created a new app with vcpkg manifest and linked this package. I also use liburing.pc file with pkg-config to add cmake command `link_directories`. But It still makes `-luring` not found error. How to resolve it?

I resolve it to add `link_directories` at root CMakeLists.txt. I'm sorry my cmake skills are not good...

**Comment by MonicaLiu0311** [comment]
The usage test passed on `x64-linux` (header files found):
```
netcpp provides CMake targets:

    find_package(netcpp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE netcpp::netcpp)
```

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by index1207** [comment]
Thanks for the review

File: ports/netcpp/portfile.cmake [hunk 1]
```diff
@@ -4,15 +4,19 @@ vcpkg_from_github(
...
     HEAD_REF release
+    PATCHES
+        pkgconfig.patch
 )
 
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DINCLUDE_TEST=OFF
```
Comment (dg0yt) [inline ports/netcpp/portfile.cmake:17]:
```suggestion
if(VCPKG_TARGET_IS_LINUX)
    vcpkg_find_acquire_program(PKGCONFIG)
endif()  

vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DINCLUDE_TEST=OFF
        "-DPKG_CONFIG_EXECUTABLE=${PKGCONFIG}"
    MAYBE_UNUSED_VARIABLES
        PKG_CONFIG_EXECUTABLE
```

File: ports/netcpp/pkgconfig.patch [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
++target_include_directories(${PROJECT_NAME} PUBLIC
+         $<BUILD_INTERFACE:${INCLUDES}>
+         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
++
++if (MSVC)
++    target_link_libraries(${PROJECT_NAME} PRIVATE ws2_32)
++else ()
++    link_directories(${URING_LIBRARY_DIRS})
++    target_include_directories(${PROJECT_NAME} PUBLIC ${URING_INCLUDE_DIRS})
++    target_link_libraries(${PROJECT_NAME} PUBLIC ${URING_LIBRARIES})
```
Comment (dg0yt) [inline ports/netcpp/pkgconfig.patch:32]:
```suggestion
+    
+    target_include_directories(${PROJECT_NAME} PRIVATE ${URING_INCLUDE_DIRS})
+    target_link_libraries(${PROJECT_NAME} PRIVATE ${URING_LINK_LIBRARIES})
```

File: ports/netcpp/vcpkg.json [hunk 3]
```diff
@@ -14,6 +18,9 @@
       "name": "vcpkg-cmake-config",
       "host": true
     },
-    "winsock2"
+    {
+      "name": "winsock2",
+      "platform": "windows"
```
Comment (dg0yt) [inline ports/netcpp/vcpkg.json:23]: I would suggest to get rid of `winsock2`. There is no added value. `netcpp` is one of two ports that have this dependency, but `ws2_32` is widely used.

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**